jep/0000/README.adoc

```diff
@@ -0,0 +1,469 @@
+                    "signature" : "somechecksumforthefile"
+                }
+            }
+        ]
+    },
+    "client" : {
+        "uri" : "https://update-cdn.example.com/some/path/to/a/evergreen-client.tar.gz",
+        "checksum" : {
+            "type" : "sha256",
+            "signature" : "somechecksumforthefile"
+        }
+    }
+}
+----
+
+The four primary keys of the update manifest are:
+
+* `meta` is an object which contains information about the instance's update
+  cycle itself, such as the `channel` and `level`.of the enclosed manifest.
+* `core` which indicates that a new jenkins.war is necessary.
+* `plugins` which will include a list of `updates` for plugins. This is an
+  object within the JSON structure rather than a flat array as it is expected
+  that at some point in the future we  may require a `removes` list to properly
+  unpublish legacy or out-dated plugins from instances.
+* `client` which indicates a new tarball for upgrading the `evergreen-client`
+  itself.
+
+Additional keys should be ignored by clients not supporting them to allow the
+Update Manifest to safely include things which are not yet supported.
+
+[NOTE]
+====
+There _may_ be opportunities to cache the Update Manifest in the future, but
+this is considered a potential optimization which will be contingent on
+observation of real world usage for Jenkins Essentials.
+====
+
+
+==== Checksums
+
+The checksums provided in the Update Manifest are not generated or validated by
+Jenkins Essentials but rather the Artifactory instance from which plugin, core,
+and other binaries are pulled.
+
+In essence, every `foo-1.2.3.hpi` has a corresponding `foo-1.2.3.hpi.sha256`
+file, the contents of which will be included as the checksum in the Update
+Manifest to enable clients to perform archive integrity validation.
+
+
+=== Client Update Behavior
+
+The client must perform the necessary downloading of items referenced in the
+<<update-manifest>> and perform checksum validation before initiating a client
+update process. The exact sequence of events and what machinery must execute
+on the client is considered outside of the scope of this document.
+
+The client should also post a new <<version-manifest>> once an update lifecycle
+successfully completed to ensure that subsequent update check-ins result in
+accurate generated Update Manifest.
+
+[[version-manifest]]
+=== Version Manifest
+
+A version manifest is the symmetrically opposite of the <<update-manifest>> in
+that it should include the actual versions of software present on a Jenkins
+Essentials instance. This may include software which is outside of the update
+lifecycle.
+
+The purpose of the purpose of the version manifest is primarily for the client
+to report to the serve a fairly accurate state of the installed software in
+the instance.
+
+
+.Version Manifest
+[source,json]
+----
+{
+    "schema" : 1,
+    "container" : {
+        "commit" : "sha1 of the built container",
```

Comment:
Image or container? Also, how do we store this inside the container, seems like a chicken-and-egg issue to me: adding a step in the Dockerfile to store a SHA would by definition add another layer?